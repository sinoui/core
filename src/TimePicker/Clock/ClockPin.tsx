import styled from 'styled-components';
import { CLOCK_PIN_SIZE } from './constants';

const ClockPin = styled.div`
  top: calc((100% - ${CLOCK_PIN_SIZE}px) / 2);
  left: calc((100% - ${CLOCK_PIN_SIZE}px) / 2);
  width: ${CLOCK_PIN_SIZE}px;
  height: ${CLOCK_PIN_SIZE}px;
  position: absolute;
  border-radius: 50%;
  background-color: ${(props) => props.theme.palette.primary.main};
`;

export default ClockPin;
