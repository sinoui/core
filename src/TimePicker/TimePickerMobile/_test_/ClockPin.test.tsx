import React from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import renderer from 'react-test-renderer';
import ClockPin from '../ClockPin';

describe('快照测试', () => {
  it('ClockPin', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <ClockPin />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
