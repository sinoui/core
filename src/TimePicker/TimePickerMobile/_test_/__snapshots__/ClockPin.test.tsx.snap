// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 ClockPin 1`] = `
<div
  className="sc-AxjAm jxcElh"
/>
`;
